lighthouse-core/lib/lh-error.js

```diff
@@ -96,6 +103,74 @@ class LighthouseError extends Error {
     const error = new Error(`Protocol error ${errMsg}`);
     return Object.assign(error, {protocolMethod: method, protocolError: protocolError.message});
   }
+
+  /**
+   * A JSON.stringify replacer to serialize LHErrors and (as a fallback) Errors.
+   * Returns a simplified version of the error object that can be reconstituted
+   * as a copy of the original error at parse time.
+   * @param {unknown} err
```

Comment:
why not a Error|LighthouseError ? curious.

Reply:
> why not a Error|LighthouseError ? curious.

that's probably better. I was originally writing it more generalized but it wasn't helpful at all